pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -1019,6 +1023,10 @@ func (c *Controller) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod, da
 		if validateErr := errors.Join(validateErrors...); validateErrors != nil {
 			return common.NewSyncError(fmt.Errorf("failed create validation: %v", validateErr), "FailedCreateValidation")
 		}
+		if downwardmetrics.IsDownwardMetricsConfigurationInvalid(c.clusterConfig, &vmi.Spec) {
+			c.recorder.Eventf(vmi, k8sv1.EventTypeWarning, controller.FeatureNotEnabled, "DownwardMetrics feature not enabled")
```

Comment:
Should this Event also occur when trying to create the VMI from VM?

Reply:
I've implemented this, but it produces a very similar event to the VMI one:
```
5s          Warning   FeatureNotEnabled         virtualmachine/vmi-fedora-1   DownwardMetrics feature not enabled
5s          Normal    SuccessfulCreate          virtualmachine/vmi-fedora-1           Started the virtual machine by creating the new virtual machine instance vmi-fedora-1
2s          Warning   FeatureNotEnabled         virtualmachineinstance/vmi-fedora-1   DownwardMetrics feature not enabled
```
The workflow is: with downwardMetrics disabled, create a VM and starting it. I'm not sure if there's another workflow in which this event may be unique.

Reply:
> 5s          Normal    SuccessfulCreate          virtualmachine/vmi-fedora-1           Started the virtual machine by creating the new virtual machine instance vmi-fedora-1

IMO no VMI should be created and this event should not occur if we know the VMI won't be able to start.

Reply:
Yeah. I agree. Left the event and now the VMI is not created.